### PR TITLE
Add support for refreshing IGMs.

### PR DESCRIPTION
**Pull request by upodroid**
Ref: https://github.com/hashicorp/terraform-provider-google/issues/5654

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**NOTE**: The compute library needs to be bumped up in the provider repos. A recently released field isn't available in the current one.

This PR implements the applyUpdatetoInstances method on IGMs. Google doesn't support setting most distruption allowed under the update_policy block so this method is called when update_policy.type is set to OPPORTUNISTIC, it is not useful to run this api call for PROACTIVE as the IGM does the update immediately.

How do I handle the new fields in terms of importing the resource? I set ImportStateVerify to false to pass the tests but what do i do with those fields?

If the approach is great, i'll apply it to the zonal IGMs.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added support for running apply instances on `compute_region_instance_group_manager`
```

### CODE REVIEW

File: third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb [hunk 1]
```diff
@@ -299,6 +299,34 @@ func resourceComputeRegionInstanceGroupManager() *schema.Resource {
 					},
 				},
 			},
+			"apply_updates_to_instances": {
```
Comment (ndmckinley) [inline third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb:302]: We've been thinking about how to represent imperative APIs like this - I'm not sure it fits well into Terraform, can you give me a short overview of the user's experience of using this block?

If I understand right, I think this could work, but I'd like to be sure I follow.

Comment (upodroid) [reply third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb:302]: The way you apply updates to instances managed via an IGM is particularly poor. You can set update policy to PROACTIVE and Google at minimum restarts the instance to apply the change of the template. If the template is changing something like network tags or metadata that doesn't need a reboot/replace which is why Google introduced the refresh option. Also, with PROACTIVE there is no way of controlling the maximum disruption an IGM update can do which is annoying. 

Implementing this introduces new options:
- Limit the IGMs most allowed disruption when updating instances you don't want to replace.
- Update fields on managed instances without RECREATE/RESTART for fields can be done such as labels, metadata etc.

Comment (upodroid) [reply third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb:302]: Any updates? I would like to move forward with it.

Comment (ndmckinley) [reply third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb:302]: We discussed this today in our meeting and I don't think we're comfortable supporting imperative actions like this right now.  We do recognize that this is valuable and we're thinking about how to support your use case, which we recognize is valid.

In particular, we are concerned about the way this update works - it triggers unconditionally if any other field is updated, but not on every `apply` (there is no way to do this in Terraform).  We think that's not very intuitive.

Comment (upodroid) [reply third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb:302]: Couple of thoughts:
- The API call needs to be executed only when the version.instance_template field is changed.
- Compute Engine Product Team should have incorporated this update method as part of the patch method.

I work at a Google Cloud Partner so i'll reach to out to my contact and see if the GCE Product Team can rework their APIs.

Comment (upodroid) [reply third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb:302]: I have updated the logic to trigger when version block is changed. `version.0.instance_template` won't work as you can specify multiple versions.

Let me know if it is good to go and i'll update the zonal igm to have this feature too.

Comment (ndmckinley) [reply third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb:302]: Great!  This looks great.  I have one naming quibble - can you put this block as `update_policy.opportunistic_update_settings` or something like that?

**Comment by ndmckinley** [comment]
All right.  This is fairly complex, I have explained it here: https://github.com/hashicorp/terraform-provider-google/issues/5654#issuecomment-692388168.

tl;dr: **We don't want to call this endpoint because it is not coherent with the goals of an `OPPORTUNISTIC` update.**
